src/config.rs

```diff
@@ -0,0 +1,67 @@
+// config.rs
+
+// *************************************************************************
+// * Copyright (C) 2020 Daniel Mueller (deso@posteo.net)                   *
+// *                                                                       *
+// * This program is free software: you can redistribute it and/or modify  *
+// * it under the terms of the GNU General Public License as published by  *
+// * the Free Software Foundation, either version 3 of the License, or     *
+// * (at your option) any later version.                                   *
+// *                                                                       *
+// * This program is distributed in the hope that it will be useful,       *
+// * but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+// * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+// * GNU General Public License for more details.                          *
+// *                                                                       *
+// * You should have received a copy of the GNU General Public License     *
+// * along with this program.  If not, see <http://www.gnu.org/licenses/>. *
+// *************************************************************************
+
+use std::fs;
+use std::path;
+
+use crate::args;
+use crate::Result;
+
+/// The configuration for nitrocli, usually read from configuration files and environment
+/// variables.
+#[derive(Clone, Copy, Debug, Default, PartialEq, serde::Deserialize)]
+pub struct Config {
+  /// The model to connect to.
+  pub model: Option<args::DeviceModel>,
+  /// Whether to bypass the cache for all secrets or not.
+  #[serde(default)]
+  pub no_cache: bool,
+  #[serde(default)]
+  /// The log level.
+  pub verbosity: u8,
+}
+
+impl Config {
+  pub fn load() -> Result<Self> {
+    load_user_config().map(|o| o.unwrap_or_else(Default::default))
+  }
+
+  pub fn update(&mut self, args: &args::Args) {
```

Comment:
Exactly.  My `Merge` implementation only merges data structures of the same type.  We could just re-use the `Args` struct for the configuration, but that has other drawbacks, see the PR text.

Reply:
> If we want to restrict the configuration to the top-level options, we could refactor these options into a `GlobalArgs` struct. Then `Args` could use this struct with `#[structopt(flatten)]`, the configuration could be deserialized into this struct too and we could automatically merge the configuration into the CLI arguments using the derived `Merge` trait.
> 
> This would allow us to remove the manual merge code. The drawback is that we would have to make bigger changes if we want to add configuration for subcommand options in the future. What do you think?

I think what you have is fine. We could do that in the future if we really see a need.

Reply:
Ups, that was meant to be a top level comment...